### DISCUSSION TITLE
Why Responses returns 204 HTTP Status? Is frontend responsible for handling the message from the server if there is no message?

### DISCUSSION THREAD

**Discussion by mndev222** [brocoders/nestjs-boilerplate#1708] [General]
Why Response returns 204 HTTP Status? Is frontend responsible for handling the message from the server if there is no message?
E.g:  reset password endpoint returns 204 HTTP Status Code if everything is ok:
![image](https://github.com/user-attachments/assets/639687df-c24c-4d46-8462-bc0730c4d9a1)
What are the standards to handle displaying the messages of successful requests? In this case if I have 204 http status code I am supposed to create a front-end messages for all the requests which returns 204 http status code. Is it good practice?
Is it better to return message from the web server and display it on the frontend? So I am supposed to change the status code to http 200 OK and return a message, e.g: "Password has been changed successfully." What are the standards?

You did it on purpose and I have to handle all responses with 204 on frontend (store messages in frontend)?

In addition. All the camelCase messages returned by nestjs web server have to be formatted and returned by i18n module, e.g.: incorrectPassword:
![image](https://github.com/user-attachments/assets/3298566d-7e52-4d95-bc75-956db4514258)

**Comment by Shchepotin** [comment]
@mndev222 excuse me for late responce.

> Why Response returns 204 HTTP Status?

204 also means success but this endpoint in case of success doesn't return the body.
https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/204

> Is frontend responsible for handling the message from the server if there is no message?
> In addition. All the camelCase messages returned by nestjs web server have to be formatted and returned by i18n module, e.g.: incorrectPassword:

You can find here https://github.com/brocoders/extensive-react-boilerplate how we handle all this stuff.
